### ISSUE TITLE
HtmlField cannot be converted to type TextField

### ISSUE THREAD

**Issue by mortifactor** [PiranhaCMS/piranha.core#940]
Hello.
While developing initial config I changed HtmlField to TextField in custom SiteType. After that I received this error:
`ArgumentException: Object of type 'Piranha.Extend.Fields.HtmlField' cannot be converted to type 'Piranha.Extend.Fields.TextField'.
`

The field in table Piranha_SiteFields still exists with old type (CLRType: Piranha.Extend.Fields.HtmlField) even after I changed it to TextField in code.

**Comment by tidyui** [comment]
Hi there’s no automatic conversion of old fields with the same name. The best way when you change type for an existing property is change the CLRType value of just delete the row from the database.

**Comment by mortifactor** [comment]
Yes, I did exactly what you say :) I thought may be it is a bug.
Btw it would be nice to do this automatically.

**Comment by tidyui** [comment]
We can probably figure something out as long as it’s well documented. Some field could have conversion rules but others need to be discarded. I’ll take a look if this could maybe be handled with implicit operators.

**Comment by tidyui** [comment]
The most feasible solution here will be to discard the value on model transformation if the type has changed. We can possible provide some conversion between built in fields.
